src/main/java/org/primeframework/mvc/util/CookieTools.java

```diff
@@ -70,7 +90,9 @@ public static <T> T fromCookie(String value, boolean encryptedIfOld, Encryptor e
     boolean compress = (result[3] & 0x02) == 0x02; // Second bit is compressed
     result = Arrays.copyOfRange(result, 4, result.length);
 
-    if (encrypt) {
+    if (encryptionRequired && !encrypt) {
+      throw new EncryptionException("Encryption is required to decrypt cookie but a non-encrypted cookie was presented");
```

Comment:
Do we need this new exception? Nobody is catching it, and the exception itself is fairly generic, so it doesn't seem like it adds value. 

This also isn't an expected use case, so I wouldn't create a special exception for it. Seem like we either ignore it, and debug log it, or throw an `IllegalStateException` or something like that. 

Perhaps the correct question to ask is, what do you want to occur if a user causes this state? Maybe that will help me make a more informed comment. We can also discuss further if you'd like.

Reply:
Good call. removed class in favor of `IllegalStateException`. It's intended to be caught here - https://github.com/prime-framework/prime-mvc/blob/wied03/cookie_encryption_non_breaking/src/main/java/org/primeframework/mvc/scope/BaseManagedCookieScope.java#L128